students/stellie/lesson03/mailroom.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+# Stella Kim
+# Assignment 1: Mailroom Part 1
+
+"""
+This program lists donors and their donation amounts.  A user is able to 
+choose from a menu of 3 donation-related actions: send a thank you, create 
+a report or quit the program.
+"""
+
+import sys
+
+# Lists donors and their contributions
+donor_db = [['John Smith', [10000, 5000, 1000]],
+            ['Mary Johnson', [50000]],
+            ['David Carlton', [2500, 500]],
+            ['James Wright', [500, 500, 500]],
+            ['Caroline Baker', [1000]]
+            ]
+
+# Displays menu options to user
+prompt = '\n'.join(('\nWelcome to the mail room.',
+          'Please choose from below options:',
+          '1 - Send a Thank You',
+          '2 - Create a Report',
+          '3 - Quit',
+          '>>> '))
+
+def send_thank_you():
+    # Prompts user to choose a thank you note option
+    user_response = input('\nPlease enter your full name or type "list" to view the list of donors: ')
+    if user_response == 'list':
+        view_donors()
+        send_thank_you()
+    else:
+        donor = search_db(user_response)  # calls function to search for already existing donor
+        if donor:
+            print('\nThis donor already exists in our database.')
+            donation = donation_amount()  # calls function to obtain donation amount made
+            donor[1].append(donation)  # adds donation amount to existing donor in database
+            thank_you_email(donor[0], donation)
+        else:
+            print('\nThis is a NEW donor and will be added to the database.')
+            donation = donation_amount()  # calls function to obtain donation amount made
+            new_donor = [user_response, [donation]]
+            donor_db.append(new_donor)  # adds new donor to database
+            thank_you_email(new_donor[0], donation)
```

Comment:
there is a lot of duplicated code in lines 38-48... can you think of a way to consolidate the logic?